File: app/src/main/java/com/iceteaviet/fastfoodfinder/data/remote/routing/model/MapsDirection.java
package com.iceteaviet.fastfoodfinder.data.remote.routing.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Genius Doan on 11/11/2016.
 */
public class MapsDirection implements Parcelable {
    public static final Creator<MapsDirection> CREATOR = new Creator<MapsDirection>() {
        @Override
        public MapsDirection createFromParcel(Parcel in) {
            return new MapsDirection(in);
        }

        @Override
        public MapsDirection[] newArray(int size) {
            return new MapsDirection[size];
        }
    };
    @SerializedName("routes")
    List<Route> routeList;

    protected MapsDirection(Parcel in) {
        routeList = in.createTypedArrayList(Route.CREATOR);
    }

    public List<Route> getRouteList() {
        return routeList;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeTypedList(routeList);
    }
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/data/remote/routing/model/MapsDirection.kt
package com.iceteaviet.fastfoodfinder.data.remote.routing.model

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

/**
 * Created by Genius Doan on 11/11/2016.
 */
class MapsDirection : Parcelable {
    @SerializedName("routes")
    var routeList: List<Route>

    constructor() {
        routeList = ArrayList()
    }

    constructor(`in`: Parcel) : this() {
        routeList = `in`.createTypedArrayList(Route.CREATOR) ?: ArrayList()
    }

    constructor(routeList: List<Route>) : this() {
        this.routeList = routeList
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeTypedList(routeList)
    }

    companion object CREATOR : Parcelable.Creator<MapsDirection> {
        override fun createFromParcel(parcel: Parcel): MapsDirection {
            return MapsDirection(parcel)
        }

        override fun newArray(size: Int): Array<MapsDirection?> {
            return arrayOfNulls(size)
        }
    }
}
